Tests/NIOTests/TestUtils.swift

```
@@ -15,14 +15,39 @@
 @testable import NIO
 import XCTest
 
-func withPipe(_ body: (NIO.FileHandle, NIO.FileHandle) -> [NIO.FileHandle]) throws {
+func withPipe(_ body: (NIO.FileHandle, NIO.FileHandle) throws -> [NIO.FileHandle]) throws {
     var fds: [Int32] = [-1, -1]
     fds.withUnsafeMutableBufferPointer { ptr in
         XCTAssertEqual(0, pipe(ptr.baseAddress!))
     }
     let readFH = FileHandle(descriptor: fds[0])
     let writeFH = FileHandle(descriptor: fds[1])
-    let toClose = body(readFH, writeFH)
+
+    let toClose: [NIO.FileHandle]
+    do {
+        toClose = try body(readFH, writeFH)
+    } catch {
+        toClose = [readFH, writeFH]
+    }
+    try toClose.forEach { fh in
+        XCTAssertNoThrow(try fh.close())
+    }
+}
+
+func withSocketpair(_ body: (NIO.FileHandle, NIO.FileHandle) throws -> [NIO.FileHandle]) throws {
+    var fds: [Int32] = [-1, -1]
+    fds.withUnsafeMutableBufferPointer { ptr in
+        XCTAssertEqual(0, socketpair(AF_UNIX, Posix.SOCK_STREAM, 0, ptr.baseAddress!))
+    }
+    let readFH = FileHandle(descriptor: fds[0])
+    let writeFH = FileHandle(descriptor: fds[1])
+
+    let toClose: [NIO.FileHandle]
+    do {
+        toClose = try body(readFH, writeFH)
+    } catch {
+        toClose = [readFH, writeFH]
```

Similarly, can we rethrow any error caught here?